pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -129,10 +129,18 @@ func (dswp *desiredStateOfWorldPopulator) populatorLoopFunc() func() {
 	}
 }
 
+func isPodTerminated(pod *api.Pod) bool {
+	return pod.Status.Phase == api.PodFailed || pod.Status.Phase == api.PodSucceeded
+}
+
 // Iterate through all pods and add to desired state of world if they don't
 // exist but should
 func (dswp *desiredStateOfWorldPopulator) findAndAddNewPods() {
 	for _, pod := range dswp.podManager.GetPods() {
+		if isPodTerminated(pod) {
+			// Do not (re)add volumes for terminated pods
```

Comment:
I thought about this but I couldn't think of a good reason to add a volume back to a node for a pod that is terminated.  The only reason I'm filtering the ones we remove is to not make waves for the release.  In general, I can't think of a reason we would want to leave volumes attached to a node for a terminated pods except for debugging.